gateway/ln-gateway/src/client.rs

```diff
@@ -15,6 +15,40 @@ use tracing::{debug, warn};
 
 use crate::{LnGatewayError, Result};
 
+pub trait IDbFactory: Debug {
+    fn create_database(&self, federation_id: FederationId, _path: PathBuf) -> Result<Database>;
+}
```

Comment:
The trait should probably move into `fedimint_api` so that each DB crate can impl it and not just the ones that happen to be anticipated here.

Reply:
Tagging with a todo :)